src/HospitalModel.H

```diff
@@ -98,6 +98,12 @@ void HospitalModel<AC,ACT,ACTD,A>::treatAgents(AC& a_agents, /*!< Agent containe
             is_alive.resize(np);
             auto is_alive_ptr = is_alive.data();
 
+            Gpu::DeviceVector<int> flag_death, flag_hosp;
+            flag_death.resize(np);
+            flag_hosp.resize(np);
+            auto flag_death_ptr = flag_death.data();
```

Comment:
Done! But with a nuance: the new `flag_status` uses the value of `1`,`2`,`3` to indicate discharge from regular hospitalization, ICU, and ventilator, respectively, while the sign indicates recovery (+) or death (-).

Consequently, I used `DiseaseStats` enum-type as you noted in [your other comment](https://github.com/AMReX-Codes/ExaEpi/pull/65/files#r1706201204), but I set the status to `DiseaseStats::<whatever> + 1` (since `DiseaseStats::hospitalization` is `0`, so can't make it negative).